mesonbuild/compilers/cpp.py

```diff
@@ -187,16 +187,52 @@ class _StdCPPLibMixin(CompilerMixinBase):
 
     """Detect whether to use libc++ or libstdc++."""
 
-    def language_stdlib_only_link_flags(self, env: 'Environment') -> T.List[str]:
+    def __init__(self) -> None:
+        self.__stdlib_flags: T.Optional[T.List[str]] = None
```

Comment:
Yeah, every instance would have a shadow copy since it's initialized to None, but that's not very idiomatic when we're not actually using the class value. I guess I could use `functools.lru_cache`

Reply:
I mean, it's not exactly annotated as a `typing.ClassVar`...

Reply:
I went with `functools.lru_cache`, we tend to use that anyway and then we don't need the helper or the private variable